webservices/react-testing.md

```diff
@@ -0,0 +1,431 @@
+# React Testing
+
+## General Guidelines
+* Use Jest for testing.
+* Use [enzyme](https://github.com/airbnb/enzyme). Furthermore, prefer using [shallow rendering](https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md).
+
+## Redux Testing
+Redux is comprised of regular JavaScript functions, so testing them is nothing special. Regardless, the following are some guidelines that will help when writing tests.
+
+### Action Creators
+Test that when the action creator is invoked with its arguments (if any) it produces the an action with the correct type and payload (if it has one).
+
+### Reducers
+Test that given an initial state and a given action it produces the correct next state. It's a good idea to add an assertion that the next state object is not the same as the initial object using the `toBe` Jest matcher.
+
+### Selectors
+Test that the selector returns the desired data from the state given the state and props (if given).
+
+## Component Testing
+
+### Testing Stateless Functional Components
+Let's say we are testing a `MealEvent` component that which may have but is not limited to the following states:
```

Comment:
Let's say we are testing a `MealEvent` component which may have, but is not limited to, the following states:

Removes "that" before the which, adds commas

Reply:
commas optional